presto-main/src/main/java/io/prestosql/execution/buffer/ArbitraryOutputBuffer.java

```diff
@@ -228,11 +233,29 @@ public void enqueue(List<SerializedPage> pages)
         masterBuffer.addPages(serializedPageReferences);
 
         // process any pending reads from the client buffers
-        for (ClientBuffer clientBuffer : safeGetBuffersSnapshot()) {
+        clientBufferPollLoop();
```

Comment:
I'd prefer to keep this inline instead of a separate method.

Reply:
Sure, placed back inline.